src/components/dock/dock.tsx

```diff
@@ -0,0 +1,177 @@
+import {
+    Component,
+    Event,
+    EventEmitter,
+    h,
+    Host,
+    Prop,
+    State,
+} from '@stencil/core';
+import { DockItem } from './dock.types';
+
+const DEFAULT_MOBILE_BREAKPOINT = 700;
+
+/**
+ * @exampleComponent limel-example-dock-basic
+ * @exampleComponent limel-example-dock-custom-component
+ * @exampleComponent limel-example-dock-mobile
+ * @exampleComponent limel-example-dock-expanded
+ * @exampleComponent limel-example-dock-colors-css
+ * @private
+ */
+@Component({
+    tag: 'limel-dock',
+    shadow: true,
+    styleUrl: 'dock.scss',
+})
+export class Dock {
+    /**
+     * Items that are placed in the dock.
+     */
+    @Prop()
+    public dockItems: DockItem[] = [];
+
+    /**
+     * Items that are placed at the bottom of the dock. (Or at the end in mobile
+     * layout.)
+     */
+    @Prop()
+    public dockFooterItems?: DockItem[] = [];
+
+    /**
+     * A label used to describe the purpose of the navigation element inside the
+     * dock to users of assistive technology, like screen readers.
+     * Example value: "Main Site Navigation"
```

Comment:
```suggestion
     * A label used to describe the purpose of the navigation element
     * to users of assistive technologies, like screen readers.
     * Specially useful when there are multiple navigation elements in the user interface.
     * Example value: "Primary navigation"
```

Reply:
Done. But corrected "Specially" to "Especially" 🙂